src/cmap/connection_pool_events.ts

```diff
@@ -104,17 +127,21 @@ export class ConnectionClosedEvent extends ConnectionPoolMonitoringEvent {
   /** The reason the connection was closed */
   reason: string;
   serviceId?: ObjectId;
+  name: string = CONNECTION_CLOSED;
+  error: Error | undefined;
 
   /** @internal */
   constructor(
     pool: ConnectionPool,
     connection: Pick<Connection, 'id' | 'serviceId'>,
-    reason: string
+    reason: string,
+    error?: Error
```

Comment:
will this ever be something that is not a `MongoError` subclass?

Reply:
Good catch. There shouldn't be a case where this is true.

Reply:
https://github.com/mongodb/node-mongodb-native/pull/3610#discussion_r1163110240